pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1176,9 +1171,31 @@ func (n *NetworkProvider) reconcileStandaloneVPCSubnets(ctx context.Context, log
 	return subs, nil
 }
 
-// getVPCRouteTable will return a list of route tables based on a route table tag
+func (n *NetworkProvider) getClusterRouteTables(ctx context.Context) ([]*ec2.RouteTable, error) {
+	routeTables, err := n.Ec2Api.DescribeRouteTables(&ec2.DescribeRouteTablesInput{})
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get route tables")
+	}
+
+	clusterVPC, err := getClusterVpc(ctx, n.Client, n.Ec2Api, n.Logger)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get cluster vpc")
+	}
+	var foundRouteTables []*ec2.RouteTable
+	for _, routeTable := range routeTables.RouteTables {
+		if *routeTable.VpcId == *clusterVPC.VpcId {
```

Comment:
Think a nil check on `routeTable` and `clusterVPC`, better safe than sorry.

Reply:
addressed